src/item.h

```diff
@@ -1184,56 +1183,99 @@ class item : public visitable
         bool reinforceable() const;
         /*@}*/
 
+        /**
+         * Helper function to retrieve any applicable resistance bonuses from item mods.
+         * @param dmg_type Damage type
+         * @return Amount of additional modded resistance
+         */
+        float get_clothing_mod_val_for_damage_type( damage_type dmg_type ) const;
+
+        /**
+        * Helper function to perform damage_type::NONE checks for forward-declared damage_type.
+        */
+        bool damage_type_none(damage_type dmg_type) const;
+
+        /**
+        * Helper function to perform damage_type validity check for forward-declared damage_type.
+        */
+        bool damage_type_invalid(damage_type dmg_type) const;
+
+        /**
+        * Helper function to check whether a damage_type can damage items for forward-declared
+        * damage_type.
+        */
+        bool damage_type_can_damage_items(damage_type dmg_type) const;
+
         /**
          * Resistance against different damage types (@ref damage_type).
          * Larger values means more resistance are thereby better, but there is no absolute value to
          * compare them to. The values can be interpreted as chance (@ref one_in) of damaging the item
          * when exposed to the type of damage.
+         * @param dmg_type The type of incoming damage
+         * @param to_self If this is true, it returns item's own resistance, not one it gives to wearer.
+         * @param bodypart_target The bodypart_id or sub_bodypart_id of the target bodypart.
+         * @param resist_value A supplied roll against coverage for physical attacks. A value of -1 causes rolls
+         *                     against each item material individually. For environmental type damage such as
+         *                     fire and acid, this value can be used to allow hypothetical calculations for
+         *                     environmental protection items such as gas masks.
+         */
+        /*@{*/
+
+        template<typename bodypart_target = bodypart_id>
+        typename std::enable_if_t < std::is_same<bodypart_target, bodypart_id>::value ||
+        std::is_same<bodypart_target, sub_bodypart_id>::value, float >
+        resist( const damage_type dmg_type, bool to_self = false,
```

Comment:
`std::enable_if_t<>` is just noise if you're not specializing this template further. 

You can move the code to the cpp file then use `extern template` in the header and manually instantiate in the cpp (see for example `Character::can_lift<>()`).

Reply:
This change should be incorporated now.